### DISCUSSION TITLE
获取公网IP的接口被墙了

### DISCUSSION THREAD

**Discussion by xuxiake2017** [NewFuture/DDNS#257] [Show and tell]
#默认 https://api-ipv4.ip.sb/ip 这个地址好像被墙了

可以自定义公网IP服务,解决

如
```json
{
"index4":"url:http://ip.3322.net/"
}
```

目前可供参考

*  http://ip.3322.net/
* https://httpbin.org/ip
* https://pv.sohu.com/cityjson?ie=utf-8
* http://whois.pconline.com.cn/ipJson.jsp
* http://v4.ipv6-test.com/api/myip.php
* http://myip.ipip.net/s

**最好IPv4和IPv6两个版本**
欢迎评论补充

**Comment by usattc** [comment]
同上

**Comment by NewFuture** [comment]
有稳定的ip查询服务可以评论补充一下

**Reply by y377** [reply]
你好，我做了一个多节点（10+节点）[hzip.net](https://hzip.net)，欢迎大家测试反馈，

**Comment by cnwangjihe** [comment]
可以试一试:
http://ip.3322.net/
https://httpbin.org/ip
https://pv.sohu.com/cityjson?ie=utf-8
http://whois.pconline.com.cn/ipJson.jsp

**Comment by cnwangjihe** [comment]
话说旧版本的还可以用
http://v4.ipv6-test.com/api/myip.php

**Comment by lirener** [comment]
http://myip.ipip.net/s
对 或者旧版本
http://v4.ipv6-test.com/api/myip.php
![QQ截图20200212181221](https://user-images.githubusercontent.com/19354130/74325292-4cc28780-4dc3-11ea-8610-e6268dd48db6.jpg)

**Comment by mem0rz** [comment]
> 话说旧版本的还可以用
> http://v4.ipv6-test.com/api/myip.php

北京联通返回偶尔返回错误错误地址，不知道原因。

**Comment by mem0rz** [comment]
搜狐的那个接口是最快的，长时间测试下来基本不会返回502
https://pv.sohu.com/cityjson

(淘宝的那个接口我IP3,4天没换这几天一直有用，他就给我频繁返回502绝了=，=)

响应速度也很快（比淘宝还快）
![1](https://user-images.githubusercontent.com/14276116/75124327-7b0c6500-56e9-11ea-9482-d15a7f6df748.png)


对比ipv6-test

![image](https://user-images.githubusercontent.com/14276116/74804904-8bed5d00-531c-11ea-924e-c4a3c7f704ad.png)
IPV6 PUBLIC可以继续沿用http://ipv6-test.com/api/myip.php 这个地址，不过我个人觉得IPV6PUBLIC没必要，直接读取网卡的就行。因为IPV6的NAT技术说实话是个偏门，不会有几个路由器支持的。

第一个ip3322没做多个服务器，只有江苏一台，偏远的用户返回较慢
第二个httpbin在国外，国外的用户可以试试
第三个sohu是最快的，在全国部署有服务器，我在北京测试响应返回速度只有12MS
第四个太平洋在线，全国部署有服务器，但是处理速度稍慢基本要40MS+
第五个my.ipip.net，全国部署有服务器，但是部分地区返回CF节点，不知道是加速乐的原因还是自身设置原因，并且处理速度是最慢的返回需要400MS+
> 
> 
> #默认 https://api-ipv4.ip.sb/ip 这个地址好像被墙了
> 
> 可以自定义公网IP服务,解决
> 
> 如
> 
> ```json
> {
> "index4":"url:http://ip.3322.net/"
> }
> ```
> 
> 目前可供参考
> 
>     * http://ip.3322.net/
> 
>     * https://httpbin.org/ip
> 
>     * https://pv.sohu.com/cityjson
> 
>     * http://whois.pconline.com.cn/ipJson.jsp
> 
>     * http://v4.ipv6-test.com/api/myip.php
> 
>     * http://myip.ipip.net/s
> 
> 
> **最好IPv4和IPv6两个版本**
> 欢迎评论补充

**Comment by mem0rz** [comment]
#https://github.com/NewFuture/DDNS/pull/132

**Comment by sosyz** [comment]
在尝试使用搜狐的接口时需要把 ip.py   _open 函数中的代码`decode('utf8')`改为`decode('gb2312')`...
否则会报`ERROR:root:'utf8' codec can't decode byte 0xc4 in position 71: invalid continuation byte`错误

**Comment by mem0rz** [comment]
> 
> 
> 在尝试使用搜狐的接口时需要把 ip.py _open 函数中的代码`decode('utf8')`改为`decode('gb2312')`...

这个地方我前几天已经注意到了，你看前几天的版本这个问题已经修正了。

**Comment by sosyz** [comment]
> > 在尝试使用搜狐的接口时需要把 ip.py _open 函数中的代码`decode('utf8')`改为`decode('gb2312')`...
> 
> 这个地方我前几天已经注意到了，你看前几天的版本这个问题已经修正了。

啊，我刚刚通过pip安装的，可能那面的代码还未更新。。打扰了

**Comment by NewFuture** [comment]
@liuwei1573 现在还是utf8, @cybmp3 关于ip获取部分的修改没有merge

这个地方倒是可以根据响应头指定的编码方式进行解码。默认 utf8

#150

**Comment by bestK** [comment]
建议添加异常处理，如   [sever酱报警功能](http://sc.ftqq.com/3.version)

**Comment by PHCSJC** [comment]
我来补充一个：
curl ip.sb
